### ISSUE TITLE
UseObjectInitializer throws exception in ASP.NET Project

### ISSUE THREAD

**Issue by jmarolf** [dotnet/roslyn#15076]
**Version Used**:  d15rel/15.0.25902.2

**Steps to Reproduce**:
1. Have the web workload installed
2. Download [this project](https://code.msdn.microsoft.com/ASPNET-MVC-Application-b01a9fe8/file/134759/6/ASP.NET%20MVC%20Application%20Using%20Entity%20Framework%20Code%20First.zip )
3. Open the project and do the required SQL project updated
4. Navigate to InstructorController.cs

**Expected Behavior**:
No analyzer crash

**Actual Behavior**:

```
Warning	AD0001	Analyzer 'Microsoft.CodeAnalysis.CSharp.UseObjectInitializer.CSharpUseObjectInitializerDiagnosticAnalyzer' threw an exception of type 'System.InvalidCastException' with message 'Specified cast is not valid.'.
'Exception occurred with following context:
Compilation: ContosoUniversity
SyntaxTree: C:\Users\jmarolf\Downloads\ASP.NET MVC Application Using Entity Framework Code First\C#\ContosoUniversity\Controllers\InstructorController.cs
SyntaxNode: new InstructorIndexData() [ObjectCreationExpressionSyntax]@[610..635) (22,28)-(22,53)

System.InvalidCastException: Specified cast is not valid.
   at Microsoft.CodeAnalysis.Options.OptionSet.GetOption[T](PerLanguageOption`1 option, String language)
   at Microsoft.CodeAnalysis.UseObjectInitializer.AbstractUseObjectInitializerDiagnosticAnalyzer`7.FadeOutCode(SyntaxNodeAnalysisContext context, OptionSet optionSet, List`1 matches, ImmutableArray`1 locations)
   at Microsoft.CodeAnalysis.UseObjectInitializer.AbstractUseObjectInitializerDiagnosticAnalyzer`7.AnalyzeNode(SyntaxNodeAnalysisContext context)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass42_1`1.<ExecuteSyntaxNodeAction>b__1()
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info)
```

**Comment by CyrusNajmabadi** [comment]
@Pilchie This is something to do with option deserialization.

**Comment by Pilchie** [comment]
@jasonmalinowski does this look like any of your other serializer bugs

**Comment by Pilchie** [comment]
@jasonmalinowski is this also fixed by #15502?

**Comment by jmarolf** [comment]
fixed with https://github.com/dotnet/roslyn/pull/16373
